File: client-flutter/lib/models/client_file.dart
class ClientFile {
  final String id;
  final String name;
  final String? url;
  final DateTime? date;

  ClientFile({
    required this.id,
    required this.name,
    this.url,
    this.date,
  });

  factory ClientFile.fromJson(Map<String, dynamic> json) {
    DateTime? parseDate(dynamic val) {
      if (val == null) return null;
      try {
        return DateTime.parse(val.toString());
      } catch (_) {
        return null;
      }
    }

    return ClientFile(
      id: json['id']?.toString() ?? '',
      name: json['filename']?.toString() ?? json['name']?.toString() ?? '',
      url: json['url']?.toString() ?? json['file']?.toString() ?? json['path']?.toString(),
      date: parseDate(json['created_at'] ?? json['createdAt'] ?? json['date']),
    );
  }
}

File: client-ios/SihlClient/Models/ClientFile.swift
import Foundation

/// Pendant zu `models/client_file.dart`.
struct ClientFile: Identifiable {
    let id: String
    let name: String
    let url: String?
    let date: Date?

    var hasUrl: Bool { !(url?.isEmpty ?? true) }

    init(json: [String: Any]) {
        self.id   = "\(json["id"] ?? UUID().uuidString)"
        self.name = (json["filename"] as? String) ?? (json["name"] as? String) ?? ""
        self.url  = (json["url"] as? String)
                 ?? (json["file"] as? String)
                 ?? (json["path"] as? String)

        let raw = json["created_at"] ?? json["createdAt"] ?? json["date"]
        self.date = Self.parseDate(raw)
    }

    private static func parseDate(_ val: Any?) -> Date? {
        guard let s = val.map({ "\($0)" }), !s.isEmpty, s != "null" else { return nil }
        return ISO8601DateFormatter().date(from: s)
            ?? DateFormatter.yyyyMMdd.date(from: String(s.prefix(10)))
    }
}

private extension DateFormatter {
    static let yyyyMMdd: DateFormatter = {
        let f = DateFormatter(); f.dateFormat = "yyyy-MM-dd"; return f
    }()
}
